lib/rust/ensogl/core/src/timer/interval.rs

```diff
@@ -0,0 +1,84 @@
+//! FRP bindings for browser `setInterval` timer.
+
+use crate::prelude::*;
+
+use enso_web::traits::ClosureOps;
+use enso_web::window;
+use enso_web::Closure;
+
+use std::time::Duration;
+
+/// Closure type alias for use in `setInterval` call.
+type TimerClosure = Closure<dyn FnMut()>;
+
+/// A repating timer based on `setIntevral` browser API.
+///
+/// The timer can be started or stopped at any time. The timer value
+/// is provided during the start.
+///
+/// When the timer expires, `expired` is emitted.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Interval {
+    /// Starts the timer immediately with provided interval value.
+    /// If the timer was already started, it is stopled and restarted.
```

Comment:
Please fix English sentences in the whole PR. In our handbook there is a sentence that you should use Gramarly (or other) plugin for your IDE. PR should be created without any English errors, otherwise it just wastes time of reviewer.